extensions/sql-database-projects/src/models/project.ts

```diff
@@ -1442,6 +1477,81 @@ export class Project implements ISqlProject {
 		}
 	}
 
+	/**
+	 * Adds a value to project property, separated by semicolon if
+	 * one or more values are already present in project property
+	 * @param propertyName Name of property
+	 * @param valueToAdd Value to be added to project property, values containing semicolon are not supported
+	 * @param caseSensitive Determines if value to add to collection should be compared against existing values case sensitively
+	 */
+	private async addValueToCollectionProjectProperty(propertyName: string, valueToAdd: string, caseSensitive?: boolean): Promise<void> {
+		if (valueToAdd.includes(';')) {
+			return;
+		}
+
+		let propertyValueToSet = this.evaluateProjectPropertyValue(propertyName);
+		if (propertyValueToSet === undefined) {
+			propertyValueToSet = valueToAdd;
+		} else {
+			const propertyValue = caseSensitive ? propertyValueToSet : propertyValueToSet.toLocaleLowerCase();
+			const originalValueToAdd = valueToAdd;
+			valueToAdd = caseSensitive ? valueToAdd : valueToAdd.toLocaleLowerCase();
+			if ((propertyValue.length > valueToAdd.length
+				&& (propertyValue.startsWith(`${valueToAdd};`)
+					|| propertyValue.endsWith(`;${valueToAdd}`)
+					|| propertyValue.indexOf(`;${valueToAdd};`) >= 0))
+				|| (propertyValue.length === valueToAdd.length
+					&& propertyValue === valueToAdd)) {
+				return;
+			}
+
+			propertyValueToSet = `${propertyValueToSet};${originalValueToAdd}`;
```

Comment:
Should be `valueToAdd` instead of `originalValueToAdd`, right?  Otherwise, if `caseSensitive` is false, then all the existing values get converted to lowercase, but the new value isn't.

Reply:
Yes, it should be `valueToAdd` instead of `originalValueToAdd`, you're right.  All existing values shouldn't get converted to lowercase, however.

`propertyValue` may be lowercased.
`propertyValueToSet` being referred to here should maintain the original casing from the project file.